src/utils.jl

```diff
@@ -157,3 +157,22 @@ function format_diff(
     $text_b
     """
 end
+
+function is_kwcall(signature::DataType)
```

Comment:
Perhaps it would make sense to use https://github.com/FluxML/MacroTools.jl or https://github.com/invenia/ExprTools.jl instead of rolling this on our own?

```julia
julia> ex = :(
                  function Base.f(x::T, y::T; z::Bool=false) where T
                      x + y
                  end
              );

julia> ExprTools.splitdef(ex)
Dict{Symbol, Any} with 6 entries:
  :args        => Any[:(x::T), :(y::T)]
  :body        => quote…
  :kwargs      => Any[:($(Expr(:kw, :(z::Bool), false)))]
  :head        => :function
  :whereparams => Any[:T]
  :name        => :(Base.f)

julia> MacroTools.splitdef(ex)
Dict{Symbol, Any} with 5 entries:
  :name        => :(Base.f)
  :args        => Any[:(x::T), :(y::T)]
  :kwargs      => Any[:($(Expr(:kw, :(z::Bool), false)))]
  :body        => quote…
  :whereparams => (:T,)
```

Reply:
Ah, but of course that's for expressions, not methods/signatures, so it might not be helpful.

Reply:
Yeah, these ones unfortunately do not offer what we need here. I couldn't find any *good way* to test this in other packages. Most just ignore old julia versions, where `Core.kwcall` doesn't exist, or use some regex with the function name of the kwsorter. The first is not an option here (unless we drop support for julia-pre-1.9, or have different semantics for the tests depending on the julia version). The second option relies on weirder internals, where we won't get good error messages in future changes (as regexes and strings just tell "no", not fail with some type error etc.). Thus, I think the current proposal here is the best we can currently do. I already added a try-catch with some additional debug information for people to open issues here.